library/alloc/src/collections/linked_list/tests.rs

```diff
@@ -1027,21 +1025,26 @@ fn extract_if_drop_panic_leak() {
     assert_eq!(d7.dropped(), 1);
 }
 
-#[test]
-#[cfg_attr(not(panic = "unwind"), ignore = "test requires unwinding support")]
-fn extract_if_pred_panic_leak() {
-    static mut DROPS: i32 = 0;
+macro_rules! struct_with_counted_drop {
+    ($struct_name:ident$(($elt_ty:ty))?, $drop_counter:ident $(=> $drop_stmt:expr)?) => {
+        thread_local! {static $drop_counter: Cell<i32> = Cell::new(0);}
```

Comment:
Preexisting, but any reason this was i32 not u32?

Reply:
I kept it on the thought that if things went really badly, this could perhaps become negative, but I suppose since it is only ever incremented, that was silly, so I'll change to u32.